src/app/components/SearchBar.tsx

```diff
@@ -1,21 +1,40 @@
 "use client";
 
 import { useRouter } from "next/navigation";
-import { useRef } from "react";
+import { ChangeEventHandler, useRef, useState } from "react";
+import { citySearch } from "../geodata/geodata";
 
 export default function SearchBar() {
   const router = useRouter();
   const searchBoxRef = useRef<HTMLInputElement>(null);
+  const [autoSuggestions, setAutoSuggestions] = useState<string[]>([]);
+
+  const handleInputChange = (async (event) => {
+    setAutoSuggestions(
+      (await citySearch(event.target.value)).map(
+        (cityRow) => `${cityRow.name}, ${cityRow["country code"]}`
+      )
+    );
+  }) as ChangeEventHandler<HTMLInputElement>;
 
   return (
     <div>
-      <input type="search" ref={searchBoxRef} />
-      <button
-        type="button"
-        onClick={() => router.push(`/?city=${searchBoxRef.current?.value}`)}
-      >
-        Search
-      </button>
+      <div>
+        <input type="search" ref={searchBoxRef} onChange={handleInputChange} />
+        <button
+          type="button"
+          onClick={() => router.push(`/?city=${searchBoxRef.current?.value}`)}
+        >
+          Search
+        </button>
+      </div>
+      {autoSuggestions.length > 0 ? (
+        <div className="border-1 inline-block">
+          {autoSuggestions.map((sug) => (
+            <div key={sug}>{sug}</div>
+          ))}
```

Comment:
```suggestion
          {autoSuggestions.slice(0, 10).map((sug) => (
            <div key={sug}>{sug}</div>
          ))}
```

Also, we probably don't want an unlimited list of suggestions. Maybe we should try limiting it to 10 or so.

Reply:
done, and added a prop to `<SearchBar>` to control the limit